src/main/kotlin/oop/State/SemaphoreState.kt

```diff
@@ -0,0 +1,12 @@
+package oop.State
+
+sealed class SemaphoreState {
+
+  class GreenSemaphore : SemaphoreState()
```

Comment:
There is no need for these to be singletons because they work like a Enum. If they had to have some behaviour we could transform them into `object`.